help_screen_design.md

```diff
@@ -0,0 +1,164 @@
+# Spike - Argparse and help screens
+
+In the Python CLI codebase, we use the [argparse](https://docs.python.org/3/library/argparse.html) module to parse commandline arguments. This third party modules controls the types of errors and help screens we return according to the types of inputs received.
+
+In this document, we will discuss how we can extend the argparser module for our purposes to match our intended UX. The entites that will be altered are based on the UX designs provided [here](https://ljfz3b.axshare.com/#id=yokln4&p=conjur_main_help&g=1).
+
+### Help screen
+
+By default, [`ArgumentParser`](https://docs.python.org/2/library/argparse.html#argparse.ArgumentParser) groups command-line arguments into “positional” and “optional” arguments when displaying help messages. When there is more appropriate grouping than the default naming, we can use the [`add_argument_group()`](https://docs.python.org/2/library/argparse.html#argparse.ArgumentParser.add_argument_group) method:
+
+For example, instead of the following
+
+```
+positional arguments:
+  {whoami,list,variable,policy}
+    whoami              Provides information about the user making an API request.
+...
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -v, --version         show program's version number and exit
+...
+
+```
+
+We can change it to:
+
+```
+Subcommands/Commands:
+  {whoami,list,variable,policy}
+    whoami              Provides information about the user making an API request.
+...
+
+Global options:
+  -v, --version         Show program's version number and exit
+...
+```
+
+In the main parser add the following:
+
+```
+parser = MyParser(description="...",
+									epilog="...",
+									add_help=False)
+```
+
+For all global arguments (optional, `--version`, `--debug`, etc) we can add the following
+
+```
+globals_optionals = parser.add_argument_group("Global arguments")
+globals_optionals.add_argument('-h', '--help', action='help', help="...")
+.
+.
+.
+```
+
+For all commands (positional, `variable`, `policy`) we can do the following:
+
+```
+resource_subparsers = parser.add_subparsers(dest='resource', title="Commands")
```

Comment:
Probably need to elbaborate about subpasers and its inner parsers (at least with the actual code of the addtion of `whoami` and `variable` parsers etc.)